gluten-core/src/main/scala/io/glutenproject/extension/ExpandFallbackPolicy.scala

```diff
@@ -157,6 +164,12 @@ case class ExpandFallbackPolicy(isAdaptiveContext: Boolean, originalPlan: SparkP
         transformPostOverrides.transformColumnarToRowExec(c2r)
```

Comment:
> Should we also handle this below case here? ColumnarToRowExec(_: BroadcastQueryStageExec)

We never fallback such case, if a broadcast exhcange is columnar but then we think a BHJ should fallback. So here did not match this case.

> It looks planWithColumnarToRow.transform applies the given rule recursively. But transformColumnarToRowExec also does some recursively work. Can they be combined into one? Not sure whether we can just re-use replaceWithTransformerPlan in transformPostOverrides.

It actually did not use the `TransformPostOverrides.replaceWithTransformerPlan`, just use `TransformPostOverrides.transformColumnarToRowExec` to replace vanilla c2r to native c2r. So there is no nested loop. The reason did not re-use `TransformPostOverrides.replaceWithTransformerPlan` is that, `TransformPostOverrides.replaceWithTransformerPlan` will remove some c2r but if we fallback to vanilla Spark, we should not remove them. I remember some test would fail if apply the whole `TransformPostOverrides`.